geoutils/raster/distributed_computing/multiproc.py

```diff
@@ -0,0 +1,194 @@
+    AbstractCluster,
+    ClusterGenerator,
+)
+
+
+def load_raster_tile(raster_unload: RasterType, tile: NDArrayNum) -> RasterType:
+    """
+    Extracts a specific tile (spatial subset) from the raster based on the provided tile coordinates.
+
+    :param raster_unload: The input raster from which the tile is to be extracted.
+    :param tile: The bounding box of the tile as [xmin, xmax, ymin, ymax].
+    :return: The extracted raster tile.
+    """
+    xmin, xmax, ymin, ymax = tile
+    # Crop the raster to extract the tile based on the bounding box coordinates
+    raster_tile = raster_unload.icrop(bbox=(xmin, ymin, xmax, ymax))
+    return raster_tile
+
+
+def remove_tile_padding(raster: RasterType, raster_tile: RasterType, tile: NDArrayNum, padding: int) -> None:
+    """
+    Removes the padding added around tiles during terrain attribute computation to prevent edge effects.
+
+    :param raster: The full DEM object from which tiles are extracted.
+    :param raster_tile: The raster tile with possible padding that needs removal.
+    :param tile: The bounding box of the tile as [xmin, xmax, ymin, ymax].
+    :param padding: The padding size to be removed from each side of the tile.
+    """
+    # New bounding box dimensions after removing padding
+    xmin, xmax, ymin, ymax = 0, raster_tile.height, 0, raster_tile.width
+
+    # Remove padding only if the tile is not at the DEM's edges
+    if tile[0] != 0:
+        tile[0] += padding
+        xmin += padding
+    if tile[1] != raster.height:
+        tile[1] -= padding
+        xmax -= padding
+    if tile[2] != 0:
+        tile[2] += padding
+        ymin += padding
+    if tile[3] != raster.width:
+        tile[3] -= padding
+        ymax -= padding
+
+    # Apply the new bounding box to crop the tile and remove the padding
+    raster_tile.icrop(bbox=(xmin, ymin, xmax, ymax), inplace=True)
+
+
+def map_block(
+    func: Callable[..., Any],
+    raster: RasterType,
+    tile: NDArrayNum,
+    depth: int,
+    *args: tuple[Any, ...],
+    **kwargs: dict[str, Any],
+) -> tuple[Any, NDArrayNum]:
+    """
+    Apply a function to a specific tile of a raster, handling loading and padding.
+
+    :param func: The function to apply to each tile.
+    :param raster: The input raster.
+    :param tile: The bounding box of the tile as [xmin, xmax, ymin, ymax].
+    :param depth: The padding size used to overlap tiles.
+    :param args: Additional arguments to pass to the function being applied.
+
+    :return: The processed tile and its bounding box.
+    """
+    # Load raster tile
+    raster_tile = load_raster_tile(raster, tile)
+
+    # Apply user-defined function to the tile
+    result_tile = func(raster_tile, *args, **kwargs)
+
+    # Remove padding
+    remove_tile_padding(raster, result_tile, tile, depth)
+
+    return result_tile, tile
+
+
+def map_multiproc(
+    func: Callable[..., Any],
+    raster_path: str | RasterType,
+    tile_size: int,
```

Comment:
I think we should use `chunks` instead of `tile_size` for this main function to keep naming consistent for users (see https://docs.dask.org/en/stable/array-chunks.html#specifying-chunk-shapes).
For internal functions, we can keep "tile" syntax if we want.